File: api/models/director.go
package models

import "gorm.io/gorm"

type Director struct {
	gorm.Model
	FirstName    string  `binding:"required" gorm:"type:varchar(100)"`
	LastName     string  `binding:"required" gorm:"type:varchar(100)"`
	BirthDate    string  `binding:"required" gorm:"type:varchar(175)"`
	BirthCountry string  `binding:"required" gorm:"type:varchar(155)"`
	Movies       []Movie `binding:"required" gorm:"many2many:movies_directors;"`
}

File: frontendreact/movie_page/src/model/director.ts
import { MovieModel } from "./movie";

export interface DirectorModel {
    id: string | null,
    firstName: string | null,
    lastName: string | null,
    birthDate: string | null,
    birthCountry: string | null,
    movies: Array<MovieModel> | null,
}

export interface SimpleDirectorModel {
    id: string | null,
    firstName: string | null,
    lastName: string | null,
    birthDate: string | null,
    birthCountry: string | null,
}